scalardb/src/scalardb/core.clj

```diff
@@ -96,22 +94,20 @@
 
 (defn- create-service-instance
   [test mode]
-  (when-let [config (some->> (c/live-nodes test)
-                             not-empty
-                             (create-properties test)
-                             DatabaseConfig.)]
-    (let [[module service] (condp = mode
-                             :storage [(StorageModule. config)
-                                       StorageService]
-                             :transaction [(TransactionModule. config)
-                                           TransactionService]
-                             :2pc [(TransactionModule. config)
-                                   TwoPhaseCommitTransactionService])
-          injector (Guice/createInjector (vector module))]
-      (try
-        (.getInstance injector service)
-        (catch Exception e
-          (warn (.getMessage e)))))))
+  (when-let [properties (some->> (c/live-nodes test)
+                                 not-empty
+                                 (create-properties test))]
+    (try
+      (condp = mode
+        :storage (.getStorage (StorageFactory/create properties))
+        :transaction (.getTransactionManager
+                      (TransactionFactory/create properties))
+        :2pc (let [factory (TransactionFactory/create properties)]
+               ; create two Two-phase commit transaction managers
+               [(.getTwoPhaseCommitTransactionManager factory)
+                (.getTwoPhaseCommitTransactionManager factory)]))
```

Comment:
Create two Two-phase commit transaction managers since the 2PC tests create two sub-transactions.